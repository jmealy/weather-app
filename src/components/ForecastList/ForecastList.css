.forecastContainer {
  display: flex;
  flex-direction: column;
  width: 80%;
  margin: 0 auto;
  color: #5c5c5c;
}

h1 {
  font-weight: lighter;
  font-size: 50px;
  margin-bottom: 20px;
}

.previewList {
  display: flex;
  flex-wrap: wrap;
  margin: 0 auto; 
}

ul {
  list-style-type: none;
  padding: 0;
}

.previewList li {
  cursor: pointer;
  border-radius: 10px;
  margin: 0 30px;
  padding: 20px;
}

.previewList li:hover {
  background: #F5F5F5;
}

img {
  width: 160px;
}

p {
  margin: 0;
}

.forecastDetails li {
  margin: 40px 0;
  font-size: 25px;
}