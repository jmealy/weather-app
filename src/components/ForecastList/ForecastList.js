import React, { useState, useEffect } from 'react';
import './ForecastList.css';

const ForecastList = ({ forecastData, city }) => {

  const [detailView, setDetailView] = useState(false);
  const [selectedForecast, setSelectedForecast] = useState(0);

  useEffect(() => setDetailView(false), forecastData);

  const showDetailView = forecast => {
    setDetailView(true);
    setSelectedForecast(forecast);
  }

  const renderForecastPreview = forecast => {
    if (!(forecast && forecast.weather)) return null;

    const date = new Date(forecast.valid_date);
    const formattedDate = date.toLocaleDateString('en-US', { weekday: 'long', month: 'short', day: 'numeric' });

    const iconUrl = `https://www.weatherbit.io/static/img/icons/${forecast.weather.icon}.png`;
    return (
      <>
        <img src={iconUrl} alt="weather icon" />
        <p>{formattedDate}</p>
      </>
    );
  }

  const renderSelectedForecast = forecast => {
    if (!(forecast && forecast.weather)) return null;

    return (
      <div className="forecastDetails">
        {renderForecastPreview(forecast)}
        <ul>
          <li>{city}</li>
          <li>{forecast.weather.description}</li>
          <li>min temp: {forecast.min_temp}</li>
          <li>max temp: {forecast.max_temp}</li>
          <li>humidity: {forecast.rh}</li>
        </ul>
      </div>
    );
  }

  return (
    <div className="forecastContainer">
      {
        detailView ?
          renderSelectedForecast(selectedForecast) :
          <>
            <h1>{city}</h1>
            <ul className="previewList">
              {forecastData.map((forecast, index) => (
                <li key={index} onClick={() => showDetailView(forecast)}>
                  {renderForecastPreview(forecast)}
                </li>
              ))}
            </ul>
          </>
      }
    </div>
  );
}

export default ForecastList;
