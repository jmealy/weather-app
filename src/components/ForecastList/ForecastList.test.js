import React from 'react';
import renderer from 'react-test-renderer';
import ForecastList from './ForecastList';

test('renders learn react link', () => {
  const city = 'London';
  const forecastData = [
    { valid_date: '23-04-2020', min_temp: 5 },
    { valid_date: '24-04-2020', min_temp: 6 },
    { valid_date: '25-04-2020', min_temp: 5 },
    { valid_date: '26-04-2020', min_temp: 3 },
    { valid_date: '27-04-2020', min_temp: 4 },
  ];
  const tree = renderer.create(<ForecastList forecastData={forecastData} city={city} />);
  expect(tree).toMatchSnapshot();
});
