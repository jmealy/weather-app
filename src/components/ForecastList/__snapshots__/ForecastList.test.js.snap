// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders learn react link 1`] = `
<div
  className="forecastContainer"
>
  <h1>
    London
  </h1>
  <ul
    className="previewList"
  >
    <li
      onClick={[Function]}
    />
    <li
      onClick={[Function]}
    />
    <li
      onClick={[Function]}
    />
    <li
      onClick={[Function]}
    />
    <li
      onClick={[Function]}
    />
  </ul>
</div>
`;
